### DISCUSSION TITLE
What is parsing and why do I need it?

### DISCUSSION THREAD

**Discussion by sstark** [gopasspw/gopass#2186] [Q&A] [answered]
First: Thank you for gopass, a program I use many times a day since years and I heavily (and happily) rely on it!

I understand what parsing is in general, but I am confused what is the concept of “parsing” in gopass. The help does not explain it for me:

    gopass -h | grep parsing
       --noparsing, -n             Do not parse the output. (default: false)

The difference I am experiencing when I turn “parsing” on is: 1) Some of my keys are now lower case instead of how I wrote them, 2) I get a warning about parsing being enabled along with a hint on how to turn it off.

I had a brief look at parse.go and understood that there can be several ways to interpret the contents of a secret, but I could not find any further explanations for this. I have now turned it off because I was starting to get annoyed by the warning and could not see any disadvantage yet.

How can I find more about this? And why is it enabled by default just to prompt me to turn it off? I would be happy to learn about this.

Sebastian

**Comment by AnomalRoil** [comment, marked as answer]
Hi @sstark, you can find more information on that front here: https://github.com/gopasspw/gopass/blob/master/docs/commands/show.md#parsing-and-secrets

Let us know if you need more details!

**Comment by sstark** [comment]
Thanks for the pointer, couldn't find that earlier. If I understand it correctly, I can disable parsing if I am not interested in using the "gopass show entry key" feature? And I won't lose any other functionality?

**Reply by AnomalRoil** [reply]
Correct. OTP and everything else should still work.
